src/vs/workbench/contrib/inlineChat/browser/inlineChatSessionService.ts

```diff
@@ -42,6 +42,7 @@ export interface IInlineChatSessionService {
 	onDidMoveSession: Event<IInlineChatSessionEvent>;
 	onDidStashSession: Event<IInlineChatSessionEvent>;
 	onDidEndSession: Event<IInlineChatSessionEndEvent>;
+	onDidChangeSessionEnablementStatus: Event<void>;
```

Comment:
I think this and the check (provideEnablementStatus) should be on the provider itself. That will align it with the real API and allows to the controller to check on all providers. Later, when calling `createSession` it passes the first enabled provider